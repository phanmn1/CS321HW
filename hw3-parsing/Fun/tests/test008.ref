Function
  int
  "main"
  Block
    If
      Neq
        Call
          "factorial"
          IntLit(4)
        IntLit(24)
      Block
        Return
          IntLit(1)
      Block
        Return
          IntLit(0)
Function
  int
  "factorial"
  Formal
    int
    "n"
  Block
    If
      LOr
        Eql
          Id("n")
          IntLit(0)
        Eql
          Id("n")
          IntLit(1)
      Block
        Return
          IntLit(1)
    [no else branch]
    Return
      Mul
        Id("n")
        Call
          "factorial"
          Sub
            Id("n")
            IntLit(1)
Function
  int
  "binomial"
  Formal
    int
    "n"
  Formal
    int
    "j"
  Block
    Return
      Div
        Call
          "factorial"
          Id("n")
        Mul
          Call
            "factorial"
            Id("j")
          Call
            "factorial"
            Sub
              Id("n")
              Id("j")
