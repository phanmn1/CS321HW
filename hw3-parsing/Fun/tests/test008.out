Invalid syntax at (18,42), n
